Eshop.WebApi/Controllers/CategoryController.cs

```diff
@@ -16,37 +24,37 @@ public class CategoryController : Controller
         [HttpGet]
         public List<Category> GetCategories()
         {
-            return Categories;
+            return DbContext.Categories.ToList(); ;
         }
 
         [HttpGet("{id}")]
         public Category GetCategory(int id) 
         { 
-            return Categories.First(x => x.Id == id);
+            return DbContext.Categories.First(x => x.Id == id);
         }
 
         [HttpPost]
         public Category CreateCategory(string name) 
         { 
-            var newId = Categories.Max(x => x.Id) + 1;
+            var newId = DbContext.Categories.Max(x => x.Id) + 1;
             var newCategory = new Category(newId, name);
 
-            Categories.Add(newCategory);
+            DbContext.Categories.Add(newCategory);
 
             return newCategory;
         }
 
         [HttpDelete("{id}")]
         public void DeleteCategory(int id) 
         { 
-            var categoryToBeDeleted = Categories.First(x => x.Id == id);
-            Categories.Remove(categoryToBeDeleted);
+            var categoryToBeDeleted = DbContext.Categories.First(x => x.Id == id);
+            DbContext.Categories.Remove(categoryToBeDeleted);
```

Comment:
dbContext.SaveChanges(); is missing - it is explained in the video that mutations need to confirm saving entity
mutations are - delete, put, patch, post
queries are - get

Reply:
added

Reply:
yes but you have added it only for 1 endpoint
for all mutations - delete, insert(create), update(put, patch) must be savechanges, otherwise the record will be not persisted in db just in memory and it is wrong.
you haven't test it otherwise you will spot it - we have discussed that insert does not work(because autoincrement of the id that we will solve in next video), but update and delete works - fix it and TEST IT!

Reply:
added

Reply:
I mean fixed lol